pkg/fileutils/fileutils_test.go

```diff
@@ -335,7 +327,7 @@ func TestMatches(t *testing.T) {
 		{"dir/**", "dir/dir2/file", true},
 		{"dir/**", "dir/dir2/file/", true},
 		{"**/dir2/*", "dir/dir2/file", true},
-		{"**/dir2/*", "dir/dir2/file/", false},
+		{"**/dir2/*", "dir/dir2/file/", true},
```

Comment:
Yes, that condition uses `HasPrefix` https://github.com/docker/docker/blob/master/pkg/archive/archive.go#L684 . If this is refactored the code from archive should be removed completely as users of this pkg shouldn't need to handle these special conditions.